pkg/satellite/satelliteweb/satelliteql/user.go

```diff
@@ -38,6 +41,61 @@ func graphqlUser() *graphql.Object {
 			fieldCreatedAt: &graphql.Field{
 				Type: graphql.DateTime,
 			},
+			companyType: &graphql.Field{
+				Type: types.Company(),
+				Resolve: func(p graphql.ResolveParams) (interface{}, error) {
+					user, _ := p.Source.(satellite.User)
+					return service.GetCompany(p.Context, user.ID)
+				},
+			},
+		},
+	})
+}
+
+// graphqlUserInput creates graphql.InputObject type needed to register/update satellite.User
+func graphqlUserInput(types Types) *graphql.InputObject {
+	return graphql.NewInputObject(graphql.InputObjectConfig{
+		Name: userInputType,
+		Fields: graphql.InputObjectConfigFieldMap{
+			fieldEmail: &graphql.InputObjectFieldConfig{
+				Type: graphql.String,
+			},
+			fieldFirstName: &graphql.InputObjectFieldConfig{
+				Type: graphql.String,
+			},
+			fieldLastName: &graphql.InputObjectFieldConfig{
+				Type: graphql.String,
+			},
+			fieldPassword: &graphql.InputObjectFieldConfig{
+				Type: graphql.String,
+			},
+			companyType: &graphql.InputObjectFieldConfig{
+				Type: types.CompanyInput(),
+			},
 		},
 	})
 }
+
+// UserInput holds info needed to register/update user
+type UserInput struct {
+	User    satellite.User
+	Company satellite.Company
+}
+
+// fromMapUser creates satellite.User from input args
+func fromMapUser(args map[string]interface{}) (input UserInput) {
```

Comment:
Let's have `fromMapUser` and `fromMapCompany` return an error if something with typecasting goes wrong. The error should be descriptive enough to know which field failed the conversion. This make easier debugging future issues.